parasol/pytest_plugin.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import pytest
+
+try:
+    import django
+    from django.conf import settings
+    from django.test import override_settings
+except ImportError:
+    django = None
+
+from parasol import django
+from parasol.schema import SolrSchema
+
+
+logger = logging.getLogger(__name__)
+
+
+if django:
+
+    @pytest.fixture(autouse=True, scope="session")
```

Comment:
My one question here (all else looking good) is whether we want session scope vs function. If the former, we're depending on the user to remember to clean out the index in-between tests, and that might be more efficient, but it also means that it would be easier to forget.

Reply:
I was thinking of it as conceptually analogous to django's test database, but now that I think about it that gets cleared and rebuilt for every method, doesn't it? 

I had been wondering about either another fixture to empty and clear it (we had an `empty_solr` fixture in PPA), or actually just letting the signals work for tests unless we explicitly turn them off. If we use signals, wouldn't it auto-index and delete?

It seems excessive to clear and rebuild the whole core every session, maybe we need a session fixture for full setup and teardown and then a function fixture to clear data?

Whatever we decide, do you think we need it now or can we refine as we continue to work with it?

Reply:
I think we can refine going forward, as long as we're aware of the functionality. And yes, the test db gets truncated after every method -- but it doesn't get rebuilt.

If we leave signals on, I'm not sure the DB cleanup will trigger them, because I believe it just fires a `TRUNCATE`. I think the fixture to empty and clear is the way to go when needed.

Reply:
@bwhicks So maybe would an `empty_solr` fixture that automatically removes all records after every function? Ok with you if I create an issue and we leave it out of this one?

Reply:
Absolutely.